### PR TITLE
Delay growing back the shroud on actor killed

### PR DESCRIPTION
**Pull request by huwpascoe**
For the bounty #7916

I don't have an opinion on how long the delay should be so feedback please!

### CODE REVIEW

File: OpenRA.Game/Traits/World/Shroud.cs [hunk 1]
```diff
@@ -92,7 +94,7 @@ public Shroud(Actor self, ShroudInfo info)
 			explored = new CellLayer<bool>(map);
 		}
 
-		void INotifyCreated.Created(Actor self)
+		public void Created(Actor self)
```
Comment (Mailaender) [inline OpenRA.Game/Traits/World/Shroud.cs]: Don't revert this although it is used inconsistently in the code base. https://github.com/OpenRA/OpenRA/pull/10143

Comment (huwpascoe) [inline OpenRA.Game/Traits/World/Shroud.cs]: Should I explicitly declare ITick.Tick?

**Comment by pchote** [comment]
We've already gone most of the way to implementing #7916 by generalizing the shroud code to support more than just actors (as described in the IRC quote there).  I think it would be a bad idea to take this as-is because it is much more limited than the alternative, has extra perf overheads, and puts that earlier work to waste.

Can you instead please look into replacing the `Actor` arguments to Add and Remove with a more general type, and then have actors spawn a new IEffect class on death that gives visibility for a set amount of time.

**Comment by huwpascoe** [comment]
Sorry, it's too vague an issue at that point. I'm gonna have to decline.

**Pull request closed without merging**